### PR TITLE
Enhancing Code Clarity: Comprehensive Comments for Newcomers, No Source Code Changes

### PR DESCRIPTION
**Pull request by ValentinMontmirail**
# Enhancing Code Clarity with Comprehensive Comments

This pull request focuses on making our codebase more accessible and welcoming to newcomers. Understanding that diving into a new codebase can be daunting, this PR introduces extensive comments throughout the code. The aim is to provide clear, helpful guidance and explanations, making it easier for new contributors to navigate and understand the structure and logic of our code.

## Key Highlights:

- 📝 Extensive Documentation: Each significant function, class, and module now includes detailed comments explaining their purpose, usage, and any nuances. This should help new contributors grasp the code's functionality more quickly.

- 🚫 No Source Code Alterations: It's important to note that this PR does not modify any existing source code. The integrity and functionality of the code remain untouched; the additions are purely in the form of comments.

- 🌟 Welcoming Newcomers: By enhancing the documentation within the code, we hope to lower the barrier of entry for new contributors. This effort reflects our commitment to fostering an inclusive and collaborative community.

## 🔍Request for Feedback:

I encourage all team members to review these comments and provide feedback. Your insights will be invaluable in ensuring that the documentation is as clear and helpful as possible. This is a collective effort to make our codebase more welcoming and understandable for everyone who joins our project.

## 🌱 Moving Forward:

This initiative is just a step towards making our project more newcomer-friendly. I look forward to working with all of you to continue improving our documentation and support for new contributors.

### CODE REVIEW

File: actor/actor.pb.go
Comment (nanderv) [inline actor/actor.pb.go:1]: Isn't this file generated code? As such, I would not put comments in there.

File: cluster/cluster.go [hunk 1]
```diff
@@ -10,64 +10,83 @@ import (
...
 
-	activationStrategy ActivationStrategy
+	activationStrategy ActivationStrategy // activationStrategy defines the strategy for activating actors within the cluster.
 
-	kinds []kind
+	kinds []kind // kinds is a slice of kinds (types of actors) available in the cluster.
 }
 
+// New creates and initializes a new Cluster based on the provided Config.
+// It returns a pointer to the Cluster and an error if any configuration issue occurs.
```
Comment (nanderv) [inline cluster/cluster.go:55]: Or, not and

File: cluster/cluster.pb.go [hunk 2]
```diff
@@ -1,8 +1,12 @@
+// Define the syntax version for this protobuf file, which is proto3.
+
 // Code generated by protoc-gen-go. DO NOT EDIT.
```
Comment (nanderv) [inline cluster/cluster.pb.go:3]: See this comment.

File: cluster/cluster.pb.go [hunk 3]
```diff
@@ -61,7 +66,7 @@ func (x *CID) ProtoReflect() protoreflect.Message {
 
 // Deprecated: Use CID.ProtoReflect.Descriptor instead.
 func (*CID) Descriptor() ([]byte, []int) {
-	return file_cluster_proto_rawDescGZIP(), []int{0}
+	return file_cluster_cluster_proto_rawDescGZIP(), []int{0}
```
Comment (nanderv) [inline cluster/cluster.pb.go:69]: source code change in generated code.

**Comment by nanderv** [comment]
Not my project, but commenting because I got interested on discord. I think generally this PR is larger than I'd like them. There's also comments added to generated files, which I don't think is wise to do (they'll be gone with the next generation).

Generally, I think comments are an addition to the code, and don't need to tell the full story. You don't have to explain that a factory-pattern-based function creates something or throws an error, that's evident from the code.

**Review by nanderv — COMMENTED** [review]
Not a full review, I think the comments apply to a large part of the PR

File: actor/actor.pb.go
Comment (ValentinMontmirail) [reply actor/actor.pb.go:1]: The comments get automatically generated once you comment the proto file :). So as long as the proto file is commented, every generation will contain the comments;

File: cluster/cluster.pb.go (hunk 2 above)
Comment (ValentinMontmirail) [reply cluster/cluster.pb.go:1]: Done. Removed from the proto file therefore, automatically removed here.

File: cluster/cluster.go (hunk 1 above)
Comment (ValentinMontmirail) [reply cluster/cluster.go]: True. It has been changed.

File: cluster/cluster.pb.go (hunk 3 above)
Comment (ValentinMontmirail) [reply cluster/cluster.pb.go]: Changed. I generate the files as they have been initially.

**Review by perbu — APPROVED** [review]
This is useful. @anthdm, I think we should merge this. what do you think?

In fairness, I think some of this is not really needed. A lot of the comments are imho a bit on the obvious side and it mostly describes what is being done, instead of the more important _why_ it is being done.

**Comment by anthdm** [comment]
This is hard for me to review. I don't really like comments. Comments should only be added if the logic is not straight forward.

**Comment by perbu** [comment]
@ValentinMontmirail going forward it might be useful to raise an issue and get some feedback on the proposed change before investing much time. this way the effort won't be wasted.

Maybe we can cherry pick a few of these that we can agree on being useful, bringing some clairity to the parts of the code that is a bit opaque.

**Comment by ValentinMontmirail** [comment]
No worry @perbu. Yep I think next time I will simply raise an issue and see how it get received :).
So, should I close this PR ?

**Comment by perbu** [comment]
@ValentinMontmirail I think that might be the right thing to do.

**Comment by ValentinMontmirail** [comment]
OK. Closing.

**Pull request closed without merging**